src/util/psbt/map/global.rs

```
@@ -158,6 +250,55 @@ impl Decodable for Global {
+
+                                let keys_count = pair.value.len() / 4 - 1;
+                                let mut decoder = Cursor::new(pair.value);
+                                let mut fingerprint = [0u8; 4];
+                                decoder.read_exact(&mut fingerprint[..])?;
+                                let mut path = Vec::<ChildNumber>::with_capacity(keys_count);
+                                while let Ok(index) = u32::consensus_decode(&mut decoder) {
+                                    path.push(ChildNumber::from(index))
+                                }
+                                let derivation = DerivationPath::from(path);
+                                // Keys, according to BIP-174, must be unique
+                                if xpub_map.insert(xpub, (Fingerprint::from(&fingerprint[..]), derivation)).is_some() {
+                                    return Err(encode::Error::ParseFailed("Repeated global xpub key"))
+                                }
+                            } else {
+                                return Err(encode::Error::ParseFailed("Xpub global key must contain serialized Xpub data"))
+                            }
+                        }
+                        PSBT_GLOBAL_VERSION => {
+                            // key has to be empty
+                            if pair.key.key.is_empty() {
+                                // there can only be one version
+                                if version.is_none() {
+                                    let vlen: usize = pair.value.len();
+                                    let mut decoder = Cursor::new(pair.value);
+                                    if vlen != 4 {
+                                        return Err(encode::Error::ParseFailed("Wrong global version value length (must be 4 bytes)"))
+                                    }
+                                    version = Some(Decodable::consensus_decode(&mut decoder)?);
```

We only understand version 0 PSBTs. According to BIP-174 we should throw an error if we see anything other than version 0.

> "If a parser encounters a version number it does not recognize, it should exit immediately as this indicates that the PSBT will contain types that it does not know about and cannot be ignored." -- ([Version Numbers](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#version-numbers))

---

Fixed in https://github.com/rust-bitcoin/rust-bitcoin/pull/499/commits/b84faa7f5e6f9d273a4a6cb78098f10c73594783